src/components/auth/node.tsx

```diff
@@ -26,6 +27,34 @@ export function Node(props: NodeProps) {
 
   if (isUiNodeInputAttributes(attributes)) {
     const shortName = attributes.name.replace('traits.', '')
+    const fieldName = hasOwnPropertyTs(strings.auth.fields, shortName)
+      ? (strings.auth.fields[shortName] as string)
+      : shortName
+
+    const messages = node.messages.map(({ text, id }, k) => {
+      const hasTranslatedMessage = hasOwnPropertyTs(strings.auth.messages, id)
+      const rawMessage = hasTranslatedMessage
+        ? strings.auth.messages[id as keyof typeof strings.auth.messages]
+        : text
+
+      const message = replacePlaceholders(rawMessage, {
+        // couldn't we find more elegant way to handle %reason% placeholder?
+        reason: text.replace(
+          'does not match pattern "^[\\\\w\\\\-]+$"',
+          strings.auth.usernameRules
+        ),
+        field: fieldName,
+      })
+
+      return (
+        <span
+          key={`${id}-${k}`}
```

Comment:
I think `k` would only make sense if a message would come up twice on the same comp.
Probably safe to remove it.